src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1989,8 +1989,10 @@ private IEnumerable<Diagnostic> FreezeDeclarationDiagnostics()
         {
             _declarationDiagnosticsFrozen = true;
 
-            // Also freeze that flag, as symbols bound after getting the declaration diagnostics shouldn't need to modify it
-            _needsGeneratedIsReadOnlyAttribute_IsFrozen = true;
+            // Also freeze generated attribute flags by observing them
+            // symbols bound after getting the declaration diagnostics shouldn't need to modify the flags
+            _ = NeedsGeneratedIsReadOnlyAttribute;
+            _ = NeedsGeneratedIsByRefLikeAttribute;
 
```

Comment:
I am discarding the reads by assigning to the "discard" variable "_" .

---
In reply to: [118791454](https://github.com/dotnet/roslyn/pull/19795#discussion_r118791454) [](ancestors = 118791454)